src/main/scala/jio/package.scala

```diff
@@ -91,21 +111,39 @@ package object jio extends JioFiles with DecorateAsScala with DecorateAsJava {
     def allBytes: Array[Byte]           = Files readAllBytes path
     def atime: Long                     = attributes.lastAccessTime.toMillis
     def attributes: BasicFileAttributes = Files readAttributes(path, classOf[BasicFileAttributes], NOFOLLOW_LINKS)
-    def blockCount: Long                = (attributes.size + blockSize - 1) / blockSize
-    def blockSize: Long                 = 512 // FIXME
     def delete(): Unit                  = Files delete path
     def exists: Boolean                 = Files.exists(path, NOFOLLOW_LINKS)
-    def filename: String                = path.getFileName.to_s
+    def isKnownType: Boolean            = isDirectory || isFile || isSymbolicLink
     def isDirectory: Boolean            = Files.isDirectory(path, NOFOLLOW_LINKS)
     def isFile: Boolean                 = Files.isRegularFile(path, NOFOLLOW_LINKS)
     def isSymbolicLink: Boolean         = Files isSymbolicLink path
     def ls: Vector[Rep]                 = (Files list path).toVector map asRep
-    def mediaType: MediaType            = MediaType(exec("file", "--brief", "--mime", "--dereference", to_s).stdout mkString "\n")
+    def mediaType: MediaType            = MediaType(new String(exec("file", "--brief", "--mime", "--dereference", to_s).stdout))
     def mtime: Long                     = attributes.lastModifiedTime.toMillis
     def readlink: Rep                   = asRep(Files readSymbolicLink path)
     def size: Long                      = attributes.size
     def to_s: String                    = path.toString
     def moveTo(target: Path)            = Files.move(path, target)
+
+    def permissions: PosixFilePermissions = {
+      val pfp = (Files getPosixFilePermissions (path, NOFOLLOW_LINKS)).asScala
+      import jnfa.PosixFilePermission._
+      PosixFilePermissions(
+        pfp(GROUP_READ) , pfp(GROUP_WRITE) , pfp(GROUP_EXECUTE),
+        pfp(OWNER_READ) , pfp(OWNER_WRITE) , pfp(OWNER_EXECUTE),
+        pfp(OTHERS_READ), pfp(OTHERS_WRITE), pfp(OTHERS_EXECUTE)
+      )
+    }
+
+   // This is not optimal, but follows existing code
+    import fs.Node._
+    def nodeType: fs.NodeType =
+           if (isFile        ) File
+      else if (isDirectory   ) Dir
+      else if (isSymbolicLink) Link
+      else `At some point in time I would love another approach`
```

Comment:
This partitioning of the filesystem elements is going to take place one way or another. NIO only gives us boolean-returning methods so those have to be translated into something. We can write our own enum, or bake it into the sum type which models the contents, but none of that feels like a materially different approach, so what did you have in mind?